src/util/pg.py

```diff
@@ -70,13 +72,13 @@ def _parallel_execute_serial(cr, queries, logger=_logger):
     return cnt
 
 
-if ThreadPoolExecutor is not None:
+if ThreadPoolExecutor is not None and concurrent is not None:
 
     def _parallel_execute_threaded(cr, queries, logger=_logger):
         """
-        Execute queries in parallel
-        Use a maximum of 8 workers (but not more than the number of CPUs)
-        Side effect: the given cursor is commited.
+        Try to execute queries concurrently using `min([8, len(queries), n_cpus])` workers.
+        If concurrency issues are detected, some of the queries will be executed sequentially.
+        Side effect: the given cursor is committed.
         As example, on `**REDACTED**` (using 8 workers), the following gains are:
```

Comment:
The example table below could be removed. The REDACTED here looks awkward.

Note that if you are documenting this you could instead add docstrings below in `parallel_execute` -- the public API.

Reply:
Yes, I will take the chance to add some documentation to the public APIs.